scripts/make_s2and_mini_dataset.py

```diff
@@ -0,0 +1,141 @@
+                X_all.append(X)
+                keys_all.append(keys)
+        elif "cluster" in file_name:
+            with open(file_name) as _json_file:
+                clusters = json.load(_json_file)
+                new_clusters = {}
+                for cluster_id, v in clusters.items():
+                    new_cluster_id = f"{dataset}_{cluster_id}"
+                    new_v = {
+                        "cluster_id": new_cluster_id,
+                        "signature_ids": [f"{dataset}_{i}" for i in v["signature_ids"]],
+                        "model_version": v["model_version"],
+                    }
+                    new_clusters[new_cluster_id] = new_v
+                clusters_all.append(new_clusters)
+        elif "paper" in file_name:
+            with open(file_name) as _json_file:
+                papers = json.load(_json_file)
+                papers_all.append(papers)
+        elif "signature" in file_name:
+            with open(file_name) as _json_file:
+                signatures = json.load(_json_file)
+                new_signatures = {}
+                for signature_id, v in signatures.items():
+                    new_signature_id = f"{dataset}_{signature_id}"
+                    new_v = {
+                        "author_id": v["author_id"],  # maybe this needs to be prepended by dataset?
+                        "paper_id": v["paper_id"],
+                        "signature_id": new_signature_id,
+                        "author_info": v["author_info"],
+                    }
+                    new_signatures[new_signature_id] = new_v
+                signatures_all.append(new_signatures)
+        else:
+            print(f"WARNING: Ignoring {file_name} in {dataset}")
+
+
+# the goal is speed so we'll remove the largest blocks (x > 250 signatures)
+# also only keep 3k blocks max.
+# aminer has 32k, inspire has 15k, and kisti has 7k blocks
+signatures_all_filtered = []
+clusters_all_filtered = []
+papers_all_filtered = []
+X_all_filtered = []
+keys_all_filtered = []
+for dataset, s, c, p, X, k in zip(DATASETS, signatures_all, clusters_all, papers_all, X_all, keys_all):
+    blocks = []
+    for v in s.values():
+        blocks.append(v["author_info"]["block"])
+    vc = pd.value_counts(blocks)
+    vc = vc[vc <= 500]  # not too big
+    vc = vc.iloc[:1000]  # and not too many, but the largest ones
+    blocks_to_keep = set(vc.index.values)
+    s_filtered = {k: v for k, v in s.items() if v["author_info"]["block"] in blocks_to_keep}
+
+    # filter the clusters too
+    c_filtered = {k: v for k, v in c.items() if np.all([i in s_filtered for i in v["signature_ids"]])}
+    clusters_all_filtered.append(c_filtered)
+
+    # go back through the clusters and find the signatures we'll actually need
+    # need to do this because sometimes the block name is just... corrupted
+    # e.g. "g miller" for most signatures but "g mller" for one...
+    singnature_keys_to_keep = set()
```

Comment:
I would say typo but I see "singnature" is used consistently... is this a word?

Reply:
Oops. It is a mega-typo.